### PR TITLE
[delivery] migrate delivery module to distroless

### PR DESCRIPTION
**Pull request by artbin**
## Description

Redis image in delivery module are now based on distroless image.

## Why do we need it, and what problem does it solve?

We'd like to base our images on a distroless image.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: delivery
type: feature
summary: Redis image is based on a distroless image.
impact_level: default
```

### CODE REVIEW

**Review by shvgn — APPROVED** [review]

File: modules/000-common/images/redis-static/Dockerfile
Comment (RomanenkoDenys) [inline modules/000-common/images/redis-static/Dockerfile:1]: If you move Dockerfile with redis static from module user-authn, please rewrite it to werf and use new image in dex-authenticator deployment

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Review by name212 — CHANGES_REQUESTED** [review]
needs investigations

**Comment by artbin** [comment]
Manual testing
==============

Get argocd-redis logs:
```bash
kubectl -n d8-delivery logs argocd-redis-567d7879ff-2s9kn
```

<details>

<summary>Logs</summary>

```
1:C 28 Nov 2023 04:14:31.700 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 28 Nov 2023 04:14:31.700 # Redis version=7.0.9, bits=64, commit=86920532, modified=1, pid=1, just started
1:C 28 Nov 2023 04:14:31.700 # Configuration loaded
1:M 28 Nov 2023 04:14:31.701 * monotonic clock: POSIX clock_gettime
1:M 28 Nov 2023 04:14:31.702 * Running mode=standalone, port=6379.
1:M 28 Nov 2023 04:14:31.702 # Server initialized
1:M 28 Nov 2023 04:14:31.702 * Ready to accept connections
```
</details>

Get argocd-server logs:
```bash
kubectl -n d8-delivery logs argocd-server-75c7796d7-lvmtv
```

<details>

<summary>Logs</summary>

```
time="2023-11-28T04:14:41Z" level=info msg="ArgoCD API Server is starting" built="2022-11-28T16:51:33Z" commit=0c7de210ae66bf631cc4f27ee1b5cdc0d04c1c96 namespace=d8-delivery port=8080 version=v2.5.3+0c7de21
time="2023-11-28T04:14:41Z" level=info msg="Starting configmap/secret informers"
time="2023-11-28T04:14:41Z" level=info msg="Configmap/secret informer synced"
time="2023-11-28T04:14:41Z" level=info msg="admin disabled"
time="2023-11-28T04:14:41Z" level=info msg="Creating client app (dex-client-argocd@d8-delivery)"
time="2023-11-28T04:14:41Z" level=info msg="argocd v2.5.3+0c7de21 serving on port 8080 (url: https://argocd.akulyabin-main.n-mitrofanov.hf.flant.com, tls: true, namespace: d8-delivery, sso: true)"
time="2023-11-28T04:14:41Z" level=info msg="Enabled application namespace patterns: d8-delivery"
time="2023-11-28T04:14:41Z" level=info msg="0xc0013291a0 subscribed to settings updates"
time="2023-11-28T04:14:41Z" level=info msg="Starting rbac config informer"
time="2023-11-28T04:14:41Z" level=info msg="RBAC ConfigMap 'argocd-rbac-cm' added"
```
</details>

Get dex-authenticator redis logs:
```bash
kubectl -n dex logs my-cool-app-dex-authenticator-d8c45859-w2vxb redis
```

<details>

<summary>Logs</summary>

```
1:C 28 Nov 2023 04:10:09.590 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 28 Nov 2023 04:10:09.590 # Redis version=7.0.9, bits=64, commit=86920532, modified=1, pid=1, just started
1:C 28 Nov 2023 04:10:09.590 # Configuration loaded
1:M 28 Nov 2023 04:10:09.591 * monotonic clock: POSIX clock_gettime
1:M 28 Nov 2023 04:10:09.592 * Running mode=standalone, port=6379.
1:M 28 Nov 2023 04:10:09.592 # Server initialized
1:M 28 Nov 2023 04:10:09.592 * Ready to accept connections
```
</details>

Get dex-authenticator logs:
```bash
kubectl -n dex logs my-cool-app-dex-authenticator-d8c45859-w2vxb dex-authenticator
```

<details>

<summary>Logs</summary>

```
[2023/11/28 04:10:10] [proxy.go:83] mapping path "/dev/null" => file system "/dev/null"
[2023/11/28 04:10:10] [oauthproxy.go:166] OAuthProxy configured for OpenID Connect Client ID: my-cool-app-dex-dex-authenticator
[2023/11/28 04:10:10] [oauthproxy.go:172] Cookie settings: name:_oauth2_proxy secure(https):true httponly:true expiry:720h0m0s domains: path:/ samesite: refresh:after 10m0s
```
</details>

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

File: modules/000-common/images/redis-static/Dockerfile
Comment (artbin) [reply modules/000-common/images/redis-static/Dockerfile:1]: done

**Pull request merged**